spark/src/test/scala/org/apache/spark/sql/delta/DeltaProtocolVersionSuite.scala

```diff
@@ -2422,18 +2407,28 @@ trait DeltaProtocolVersionSuiteBase extends QueryTest
            |$featureProperty = "true"
            |)""".stripMargin)
 
-    val readerVersion = Math.max(feature.minReaderVersion, 1)
-    val expectedWriterFeatures =
-      Some(Set(feature.name, InvariantsTableFeature.name, AppendOnlyTableFeature.name))
-    val expectedReaderFeatures: Option[Set[String]] =
-      if (supportsReaderFeatures(readerVersion)) Some(Set(feature.name)) else None
+    /*
+    val expectedReaderVersion = Math.max(feature.minReaderVersion, 1)
+    val expectProtocol = if (feature.isLegacyFeature) {
+      Protocol(expectedReaderVersion, feature.minWriterVersion)
+    } else {
+     Protocol(
+       expectedReaderVersion,
+       TABLE_FEATURES_MIN_WRITER_VERSION,
+       if (supportsReaderFeatures(expectedReaderVersion)) Some(Set(feature.name)) else None,
+       Some(Set(feature.name, InvariantsTableFeature.name, AppendOnlyTableFeature.name)))
+    }
+    */
+    val expectProtocol = if (feature.isLegacyFeature) {
+      Protocol(Math.max(1, feature.minReaderVersion), feature.minWriterVersion)
+    } else {
+      Protocol(3, 7).withFeatures(Seq(
+        InvariantsTableFeature,
+        AppendOnlyTableFeature,
+        feature)).normalized
```

Comment:
can we be explicit about which protocol we expect and not use normalized please?